### PR TITLE
reimplement sidecar with good design and ease of use

### PR DESCRIPTION
**Pull request by chenshanpei**
The origninal sidecar mechanism has many drawbacks:

      1. Hard to use. The developer need to manually copy sidecar functions
       and their dependencies code. Also, it doesn't report error when the
       sidecar functions are optimized by GCC.

    2. Very bad design. It introduces two redundant files xxx_sidecar.h,
       which makes code complex. To make matters worse, it adds a time-consuming
       step in the build stage.

### CODE REVIEW

File: boundary/analyze.py [hunk 1]
```diff
@@ -145,22 +147,51 @@ def lookup_if_global(name_and_file):
     file = global_fn_dict.get(name, None) if file == '?' else file
     return (name, file) if file else None
 
+def sidecar_inflect(sidecar, in_vmlinux):
+    assert len(sidecar - in_vmlinux) == 0, \
```
Comment (ampresent) [inline boundary/analyze.py:151]: `assert not (sidecar - in_vmlinux), \` is okay too

File: boundary/analyze.py [hunk 2]
```diff
@@ -145,22 +147,51 @@ def lookup_if_global(name_and_file):
...
+            'sidecar functions should not be optimzied by GCC'
+
+    insiders = set()
```
Comment (ampresent) [inline boundary/analyze.py:154]: The term `insider` means the function is updated implicitly, right after all interface function are redirected.

However this is not the case for these descendants of sidecar functions. Replacing one sidecar function doesn't guarantee its descendants gets replaced as well.

So a better name is needed here.

**Review by ampresent — COMMENTED** [review]
Others LGTM

File: boundary/analyze.py (hunk 2 above)
Comment (chenshanpei) [reply boundary/analyze.py]: rename to `leftover`

**Pull request merged**